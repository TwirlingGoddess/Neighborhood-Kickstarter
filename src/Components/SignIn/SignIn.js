import React, { Component } from 'react'
import GoogleLogin from 'react-google-login';
import { getUser, getAllUsers } from '../../utilities/apiCalls/apiCalls'
import './SignIn.css'

class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      userName: '',
      password: '',
      allUsers: []
    }
  }

  componentDidMount = () => {
    this.setUsers();
  }

  setUsers = async () => {
    let allUsers = await getAllUsers();
    console.log(allUsers)
    this.setState({
      allUsers
    })
  }

  getCurrentUser = (token) => {
    let foundUser = this.state.allUsers.find(user => {
      console.log(token)
      console.log(user.token)
      return user.token === token
    })
    console.log(foundUser)
    if(foundUser) {
      this.props.updateUser(foundUser)
      this.props.history.push('/Landing')
    }
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    })
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({
      userName: '',
      password: ''
    })
  }

  responseGoogle = async (response) => {
    console.log(response.profileObj.googleId)
    let googleSignIn = {
      token: response.accessToken
    };

    this.getCurrentUser(response.profileObj.googleId)
  }

  render() {
    return (
      <div className='sign-in-section'>
        <form className='sign-in-form' onSubmit={this.handleSubmit}>
          <h1>Sign In</h1>
          <input
            className='signin-input-fields'
            name='userName'
            type='text'
            placeholder='user name'
            value={this.state.userName}
            onChange={this.handleChange}
          />
          <input
            className='signin-input-fields'
            name='password'
            type='password'
            placeholder='password'
            value={this.state.password}
            onChange={this.handleChange}
          />
          <button className='sign-in-button'>Sign In</button>
          <span>OR</span>
          <GoogleLogin
            clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
            buttonText="Login with Google"
            onClick={this.responseGoogle}
            onSuccess={this.responseGoogle}
            onFailure={this.responseGoogle}
          />
        </form>
      </div>
    )
  }
}
export default SignIn