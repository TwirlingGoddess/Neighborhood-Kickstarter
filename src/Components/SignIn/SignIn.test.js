import React from 'react';
import SignIn from './SignIn';
import GoogleLogin from 'react-google-login';
import { shallow } from 'enzyme';

describe('SignIn', () => {
  let wrapper;
  it('should match the snapshot', () => {
    wrapper = shallow(<SignIn />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state with the neighborhood selected', () => {
    let eventObject = { target: { name: 'userName', value: 'a' } };

    wrapper = shallow(<SignIn />);
    wrapper.find('.username').simulate('change', eventObject)
    expect(wrapper.state('userName')).toEqual('a');
  });

  it('should sign in with google if the google button is clicked', () => {
    let mockGetCurrentUser = jest.fn();

    wrapper = shallow(<SignIn getCurrentUser={mockGetCurrentUser}/>);
    wrapper.instance().responseGoogle();


  })
}); 