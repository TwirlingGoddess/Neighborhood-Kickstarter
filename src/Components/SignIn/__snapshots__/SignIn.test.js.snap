// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="sign-in-form"
      >
        <h1>
          Sign In
        </h1>
        <input
          className="signin-input-fields"
          name="userName"
          onChange={[Function]}
          placeholder="user name"
          type="text"
          value=""
        />
        <input
          className="signin-input-fields"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
        <button
          className="sign-in-button"
          onClick={[Function]}
        >
          Sign In
        </button>
        <span>
          OR
        </span>
        <t
          accessType="online"
          buttonText="Login with Google"
          clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
          cookiePolicy="single_host_origin"
          disabledStyle={
            Object {
              "opacity": 0.6,
            }
          }
          fetchBasicProfile={true}
          isSignedIn={false}
          jsSrc="https://apis.google.com/js/client:platform.js"
          onClick={[Function]}
          onFailure={[Function]}
          onRequest={[Function]}
          onSuccess={[Function]}
          prompt=""
          scope="profile email"
          tag="button"
          type="button"
          uxMode="popup"
        />
      </form>,
      "className": "sign-in-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Sign In
          </h1>,
          <input
            className="signin-input-fields"
            name="userName"
            onChange={[Function]}
            placeholder="user name"
            type="text"
            value=""
          />,
          <input
            className="signin-input-fields"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />,
          <button
            className="sign-in-button"
            onClick={[Function]}
          >
            Sign In
          </button>,
          <span>
            OR
          </span>,
          <t
            accessType="online"
            buttonText="Login with Google"
            clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
            cookiePolicy="single_host_origin"
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            isSignedIn={false}
            jsSrc="https://apis.google.com/js/client:platform.js"
            onClick={[Function]}
            onFailure={[Function]}
            onRequest={[Function]}
            onSuccess={[Function]}
            prompt=""
            scope="profile email"
            tag="button"
            type="button"
            uxMode="popup"
          />,
        ],
        "className": "sign-in-form",
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signin-input-fields",
            "name": "userName",
            "onChange": [Function],
            "placeholder": "user name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signin-input-fields",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "className": "sign-in-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "OR",
          },
          "ref": null,
          "rendered": "OR",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessType": "online",
            "buttonText": "Login with Google",
            "clientId": "603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com",
            "cookiePolicy": "single_host_origin",
            "disabledStyle": Object {
              "opacity": 0.6,
            },
            "fetchBasicProfile": true,
            "isSignedIn": false,
            "jsSrc": "https://apis.google.com/js/client:platform.js",
            "onClick": [Function],
            "onFailure": [Function],
            "onRequest": [Function],
            "onSuccess": [Function],
            "prompt": "",
            "scope": "profile email",
            "tag": "button",
            "type": "button",
            "uxMode": "popup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="sign-in-form"
        >
          <h1>
            Sign In
          </h1>
          <input
            className="signin-input-fields"
            name="userName"
            onChange={[Function]}
            placeholder="user name"
            type="text"
            value=""
          />
          <input
            className="signin-input-fields"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
          <button
            className="sign-in-button"
            onClick={[Function]}
          >
            Sign In
          </button>
          <span>
            OR
          </span>
          <t
            accessType="online"
            buttonText="Login with Google"
            clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
            cookiePolicy="single_host_origin"
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            isSignedIn={false}
            jsSrc="https://apis.google.com/js/client:platform.js"
            onClick={[Function]}
            onFailure={[Function]}
            onRequest={[Function]}
            onSuccess={[Function]}
            prompt=""
            scope="profile email"
            tag="button"
            type="button"
            uxMode="popup"
          />
        </form>,
        "className": "sign-in-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Sign In
            </h1>,
            <input
              className="signin-input-fields"
              name="userName"
              onChange={[Function]}
              placeholder="user name"
              type="text"
              value=""
            />,
            <input
              className="signin-input-fields"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />,
            <button
              className="sign-in-button"
              onClick={[Function]}
            >
              Sign In
            </button>,
            <span>
              OR
            </span>,
            <t
              accessType="online"
              buttonText="Login with Google"
              clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
              cookiePolicy="single_host_origin"
              disabledStyle={
                Object {
                  "opacity": 0.6,
                }
              }
              fetchBasicProfile={true}
              isSignedIn={false}
              jsSrc="https://apis.google.com/js/client:platform.js"
              onClick={[Function]}
              onFailure={[Function]}
              onRequest={[Function]}
              onSuccess={[Function]}
              prompt=""
              scope="profile email"
              tag="button"
              type="button"
              uxMode="popup"
            />,
          ],
          "className": "sign-in-form",
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signin-input-fields",
              "name": "userName",
              "onChange": [Function],
              "placeholder": "user name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signin-input-fields",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "className": "sign-in-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OR",
            },
            "ref": null,
            "rendered": "OR",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessType": "online",
              "buttonText": "Login with Google",
              "clientId": "603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com",
              "cookiePolicy": "single_host_origin",
              "disabledStyle": Object {
                "opacity": 0.6,
              },
              "fetchBasicProfile": true,
              "isSignedIn": false,
              "jsSrc": "https://apis.google.com/js/client:platform.js",
              "onClick": [Function],
              "onFailure": [Function],
              "onRequest": [Function],
              "onSuccess": [Function],
              "prompt": "",
              "scope": "profile email",
              "tag": "button",
              "type": "button",
              "uxMode": "popup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
